fix: ensure ref is present before using

STRML opened the pull request:
It's possible for the ref to be undefined, especially when using async batching strategies.

For example:

```
TypeError: Cannot read property 'getBoundingClientRect' of null
  at getSliderStart(node_modules/rc-slider/lib/createSlider.js:170:0)
  at calcValueByPos(node_modules/rc-slider/lib/createSlider.js:196:0)
  at onMove(node_modules/rc-slider/lib/Slider.js:111:0)
  at callback(node_modules/rc-slider/lib/createSlider.js:120:0)
  at batchedUpdates(ui/utils/ReactRAFBatchingStrategy.js:29:13)
  at unstable_batchedUpdates(node_modules/react-dom/lib/ReactUpdates.js:97:0)
  at call(node_modules/rc-util/lib/Dom/addEventListener.js:21:0)
  at apply(node_modules/add-dom-event-listener/lib/index.js:17:0)
  at apply(node_modules/raven-js/src/raven.js:278:0)
```

benjycui commented:
Could you add a unit test for this in https://github.com/react-component/slider/blob/master/tests/common.test.js ? THX.

STRML commented:
Updated - also noticed that these event handlers could remain in the DOM if the component is unmounted before mouseup/touchend. Added a `componentWillUnmount` and corresponding test.

benjycui commented:
6.0.2
